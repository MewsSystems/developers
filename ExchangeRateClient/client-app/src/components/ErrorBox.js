// @flow strict
import styled from "styled-components";
import * as React from "react";
import { FaExclamationTriangle } from "react-icons/fa";

const Container = styled.div`
  display: flex;
  padding: 2rem 3rem;
  color: ${({ theme }) => theme.colors.alert};
  font-weight: bold;
  font-size: 2em;
`;
const Icon = styled(FaExclamationTriangle)`
  margin-right: 20px;
`;

export default ({ error }: { error: Error }) => (
  <Container>
    <Icon /> ERROR: {error.message}
  </Container>
);
