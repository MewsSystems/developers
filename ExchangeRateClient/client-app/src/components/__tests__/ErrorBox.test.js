// @flow
import { render, cleanup } from "react-testing-library";
import React from "react";
import ErrorBox from "../ErrorBox";

afterEach(cleanup);

jest.mock("react-icons/fa", () => ({ FaExclamationTriangle: jest.fn(() => "Mocked React Icon") }));

describe("ErrorBox", () => {
  test("render", () => {
    const { baseElement } = render(<ErrorBox error={new Error("test error")} />);
    expect(baseElement).toMatchSnapshot();
  });
});
