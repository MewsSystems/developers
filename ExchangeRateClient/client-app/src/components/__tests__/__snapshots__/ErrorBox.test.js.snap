// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox render 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa dLCTwP"
    >
      Mocked React Icon
      ERROR: 
      test error
    </div>
  </div>
</body>
`;
